src/FilterBundle/Helper/EnumAttributeHelper.php

```diff
@@ -47,41 +70,51 @@ public function __construct(Client $client)
      */
     public function getLabels(string $attributeName): array
     {
-        $productTypeRequest = ProductTypeQueryRequest::of();
+        $cacheItem = $this->cacheItemPool->getItem(self::CACHE_KEY . '-' . md5($attributeName));
 
-        $response = $this->client->execute($productTypeRequest);
+        if ($cacheItem->isHit()) {
+            $labels = $cacheItem->get();
+        } else {
+            $productTypeRequest = ProductTypeQueryRequest::of();
```

Comment:
there is also a way to reduce request size by filter on attribute names. maybe thats also an option

```
    public function getFacetsArrayByName(string $field) : array
    {
        $request = new ProductTypeQueryRequest();
        $request->addParam('where', 'attributes(name="' . $field . '")');
```